privacyidea/api/validate.py

```diff
@@ -426,44 +356,61 @@ def samlcheck():
     mapping.
     """
     user = request.User
+    result = success = False
+    serial = getParam(request.all_data, "serial")
     password = getParam(request.all_data, "pass", required)
+    otp_only = getParam(request.all_data, "otponly")
     token_type = getParam(request.all_data, "type")
     options = {"g": g,
-               "clientip": g.client_ip,
-               "token_type": token_type}
+               "clientip": g.client_ip}
     # Add all params to the options
     for key, value in request.all_data.items():
             if value and key not in ["g", "clientip"]:
```

Comment:
i am not entirely sure what the intention of this construct is.
If `value` and `key` exist, the result would be to check if `key` is not in the list.

Reply:
This is an old construct and not topic of this PR.
However, if the value is not empty, then the key and value are copied to the options.
To my knowledge the intention is, to pass all parameters except "g" and "clientip" to the lower levels.

Reply:
...and also empty values were not supposed to be passed.
If we change this, empty values would be passed to the lower level and we would change the behaviour unexpectedly.